commit fc04488b427e2e9ee5e7b76836a018b062b5cd12 (HEAD -> master)
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-09-11T21:30:39+00:00 (4 weeks ago)
    
    MANTA-4535 `manta-adm update` breaks when deploying Prometheus if CNS allow_transfer metadata does not exist yet
    Reviewed by: Tim Foster <tim.foster@joyent.com>
    Approved by: Tim Foster <tim.foster@joyent.com>

diff --git a/lib/deploy.js b/lib/deploy.js
index 10c6508..8f76870 100644
--- a/lib/deploy.js
+++ b/lib/deploy.js
@@ -742,7 +742,9 @@ Deployer.prototype.addCnsAllowTransfer = function addCnsAllowTransfer(cb)
 	 * Update the cns service with the vm's admin IP, if necessary.
 	 */
 	function updateCnsSvc(ctx, next) {
-		var existingIps = ctx.cnsSvc.metadata.allow_transfer;
+		var allow_transfer = ctx.cnsSvc.metadata.allow_transfer;
+		var existingIps =
+		    allow_transfer === undefined ? [] : allow_transfer;
 		if (existingIps.indexOf(ctx.ip) > -1) {
 			log.info({
 				svcname: self.svcname,
